pilot/pkg/xds/sds.go

```diff
@@ -78,14 +84,16 @@ func sdsNeedsPush(updates model.XdsUpdates) bool {
 // Invalid resource names are ignored
 func (s *SecretGen) parseResources(names []string, proxy *model.Proxy) []SecretResource {
 	res := make([]SecretResource, 0, len(names))
+	pkpConf := proxy.Metadata.ProxyConfigOrDefault(s.meshConfig.GetDefaultConfig()).GetPrivateKeyProvider()
+	pkpConfHash := md5.Sum([]byte(pkpConf.String()))
```

Comment:
We may consider using something other than md5 here... while it should be fine, security auditors like to complain when you use md5, especially since this is in a security sensitive context. We are using xxhash in other places now, maybe just swap to that

Reply:
ref: https://github.com/istio/istio/blob/e21c457a6c2bfb6a3e387dfcce4e078c133206b3/pilot/pkg/model/config.go#L60

Reply:
Done. md5 can not even pass the CI security test.